### PR TITLE
Check chatmessages for illegal chars & moved length check into the packet class

### PR DESCRIPTION
**Pull request by Outfluencer**
If command logging is enabled the control chars could break the console and we fix the issue that hack client could use \n and color codes in commands

### CODE REVIEW

File: proxy/src/main/java/net/md_5/bungee/connection/UpstreamBridge.java [hunk 1]
```diff
@@ -144,9 +144,12 @@ public void handle(KeepAlive alive) throws Exception
     @Override
     public void handle(Chat chat) throws Exception
     {
-        int maxLength = ( con.getPendingConnection().getVersion() >= ProtocolConstants.MINECRAFT_1_11 ) ? 256 : 100;
-        Preconditions.checkArgument( chat.getMessage().length() <= maxLength, "Chat message too long" ); // Mojang limit, check on updates
-
+        for(char character : chat.getMessage().toCharArray()) {
```
Comment (terrarier2111) [inline proxy/src/main/java/net/md_5/bungee/connection/UpstreamBridge.java:147]: String allows for random access, hence you don't have to convert it to a char array.

Comment (Outfluencer) [reply proxy/src/main/java/net/md_5/bungee/connection/UpstreamBridge.java:147]: You're right i will do it with charAt

File: proxy/src/main/java/net/md_5/bungee/connection/UpstreamBridge.java [hunk 2]
```diff
@@ -146,8 +146,8 @@ public void handle(Chat chat) throws Exception
     {
         for( int index = 0; index < chat.getMessage().length(); index++ )
```
Comment (Janmm14) [inline proxy/src/main/java/net/md_5/bungee/connection/UpstreamBridge.java:147]:
```suggestion
        for( int index = 0, length = chat.getMessage().length(); index < length; index++ )
```

**Review by Janmm14 — CHANGES_REQUESTED** [review]

**Review by Janmm14 — APPROVED** [review]

**Comment by md-5** [comment]
Thanks

**Pull request closed without merging**